// Copyright 2020 Lisandro Bravo.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

// Angles of the landmarks
// TODO: replace angle1-2 for a repeated field 
// angle1: contains the angle for the particular joint (mcp pip dip)
//         except for LM0 which contains the inclination of the whole hand 
// angle2: is used to determine the angle of finger intersections, starting from the thumb
//         in example, the angle for LM:2, the base of the thumb is calculated using LM2 as the vertex and 
//         LM4 (tip of thumb) and LM8 (tip of index)


message Angle{
  optional int32 landmarkID = 1;
  optional float angle1 = 2;
  optional float angle2 = 3;
}


